### PR TITLE
TINY-8322: Cleaned up APIs that didn't require the new keyword

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-8322

Description of Changes:
* Removed the `new` keyword from the types/docs examples for anything that's not a class
* Swapped out `Styles` from being a "partial" class to keep it consistent with other APIs (DOMUtils for example)

Pre-checks:
* [x] Changelog entry added
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Aligning a table to the left or right will now use margin styling instead of float styling #TINY-6558
 - The `tinymce.settings` global property is no longer set upon initialization #TINY-7359
 - Addons such as plugins and themes are no longer constructed using the `new` operator #TINY-8256
+- A number of APIs that were not actually classes no longer require using the `new` operator #TINY-8322
```
Comment (metricjs) [inline modules/tinymce/CHANGELOG.md:24]:
```suggestion
- A number of APIs that were not classes no longer require using the `new` operator #TINY-8322
```
I suspect Tyler won't like using "actually". Could use "proper" maybe, if you want something there?

**Review by metricjs — APPROVED** [review]

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:24]: We need something there to show they were "fake"/"partial" classes and I'm not sure `proper` covers that. It's better than nothing though but I'm going to ask @tylerkelly13 to weigh in.

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:24]:
```suggestion
- A number of APIs that were not proper classes, are no longer constructed using the new operator #TINY-8322
```
Was talking to Tyler via DM's and this is the best we've come up with so far :shrug:

File: modules/tinymce/CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Aligning a table to the left or right will now use margin styling instead of float styling #TINY-6558
 - The `tinymce.settings` global property is no longer set upon initialization #TINY-7359
 - Addons such as plugins and themes are no longer constructed using the `new` operator #TINY-8256
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:23]:
```suggestion
- Add-ons such as plugins and themes are no longer constructed using the `new` operator #TINY-8256
```
Also fixing a typo from a previous PR

**Pull request merged**